pkg/lib/crd/version.go

```diff
@@ -23,24 +25,28 @@ var supportedCRDVersions = map[string]struct{}{
 
 // Version takes a CRD manifest and determines whether it is v1 or v1beta1 type based on the APIVersion.
 func Version(manifest *string) (string, error) {
+	if manifest == nil {
+		return "", fmt.Errorf("empty CRD manifest")
+	}
+
 	dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(*manifest), 10)
 	unst := &unstructured.Unstructured{}
 	if err := dec.Decode(unst); err != nil {
 		return "", err
 	}
 
-	v := unst.GetObjectKind().GroupVersionKind().Version
+	v := unst.GroupVersionKind().Version
 	if _, ok := supportedCRDVersions[v]; !ok {
-		return "", fmt.Errorf("could not determine CRD version from manifest")
+		return "", fmt.Errorf("CRD APIVersion from manifest not supported: %s", v)
 	}
 
 	return v, nil
 }
 
-// NotEqual determines whether two CRDs are equal based on the versions and validations of both.
-// NotEqual looks at the range of the old CRD versions to ensure index out of bounds errors do not occur.
+// V1NotEqual determines whether two v1 CRDs are equal based on the versions and validations of both.
+// V1NotEqual looks at the range of the old CRD versions to ensure index out of bounds errors do not occur.
 // If true, then we know we need to update the CRD on cluster.
-func NotEqual(currentCRD *apiextensionsv1.CustomResourceDefinition, oldCRD *apiextensionsv1.CustomResourceDefinition) bool {
+func V1NotEqual(currentCRD *apiextensionsv1.CustomResourceDefinition, oldCRD *apiextensionsv1.CustomResourceDefinition) bool {
```

Comment:
does this func mean that CRDs from an upgrade are not updated in the cluster if the versions are equal? What about the other 47 fields of a CRD?

Reply:
That's a good point. We have elected to remove this check entirely since `InstallPlan` resources are not continually reconciled, and so we will attempt to update the CRD on the cluster each time.